src/main/java/com/bobocode/hoverla/bring/context/BeanAnnotationClassValidator.java

```diff
@@ -0,0 +1,145 @@
+package com.bobocode.hoverla.bring.context;
+
+import com.bobocode.hoverla.bring.annotation.Inject;
+import com.bobocode.hoverla.bring.exception.BeanClassValidationException;
+import lombok.extern.slf4j.Slf4j;
+
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import static java.lang.String.format;
+import static java.lang.reflect.Array.getLength;
+
+@Slf4j
+public class BeanAnnotationClassValidator {
+
+    private static final String VIOLATION_MESSAGES_DELIMITER = "\n- ";
+
+    private static final Map<String, Predicate<Class<?>>> TYPE_PREDICATE_MAP = Map.of(
+            "Interface", Class::isInterface,
+            "Enum", Class::isEnum,
+            "Record", Class::isRecord,
+            "Abstract class", type -> Modifier.isAbstract(type.getModifiers()),
+            "Inner type ", type -> Objects.nonNull(type.getEnclosingClass())
+    );
+
+    private final List<String> validationViolations = new LinkedList<>();
+
+    public void validateBeanClasses(Set<Class<?>> beanClasses) {
+        log.info("Started validation of {} scanned `@Bean` classes", beanClasses.size());
+        for (Class<?> beanClass : beanClasses) {
+            String beanClassName = beanClass.getSimpleName();
```

Comment:
@DanielChanger well, probably…
I did not use that since I didn’t want log messages to be too long. But it makes sense to include package info, will fix